minitests/partial_reconfig_flow/Makefile

```diff
@@ -0,0 +1,83 @@
+# Top-level target for generating a programmable bitstream.  Given a .fasm
+# file, calling make with the .fasm extension replaced with .hand_crafted.bit
+# will generate a bitstream that includes both the harness and the .fasm design
+# ready for programming to a board.  For example, 'make
+# roi_noninv.hand_crafted.bit' will generate a bitstream that includes the
+# design from roi_noninv.fasm. 
+%.hand_crafted.bit: init_sequence.bit %.no_headers.bit final_sequence.bit
+	cat $^ > $@
+
+%.no_headers.bit: %.patched.bit
+	# WARNING: these values need to be tweaked if anything about the
+	# Vivado-generated design changes.
+	xxd -p -s 0x18 $< | xxd -r -p - $@
+
+%.patched.bit: %.frm harness_routed.bit
+	${XRAY_TOOLS_DIR}/xc7patch \
+		--part_file ${XRAY_PART_YAML} \
+		--bitstream_file harness_routed.bit \
+		--frm_file $< \
+		--output_file $@
+
+# xc7patch currently only generates the actual frame writes which is
+# insufficient to program a device.  Grab the initialization and finalization
+# sequences from the harness bitstream so they can be tacked on to the
+# xc7patch-generated bitstream to create a programmable bitstream.
+init_sequence.bit: harness_routed.bit
```

Comment:
This looks super fragile. I'm not entirely clear where some of these numbers are coming from. For example, if I want to target arty a7 instead of basys3 (changes harness pnr), do these all need to be changed?

Reply:
I doubt it.  I'll add an explanation of how I came up with these offsets.